src/Microsoft.AspNetCore.SystemWebAdapters.Authentication/RemoteAuthenticationModule.Framework.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Web;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.Authentication;
+
+internal sealed class RemoteAuthenticationModule : IHttpModule
+{
+    private readonly RemoteAuthenticationOptions _options;
+
+    public RemoteAuthenticationModule(RemoteAuthenticationOptions options)
+    {
+        _options = options;
+    }
+
+    public void Init(HttpApplication context)
+    {
+        var handler = new RemoteAuthenticationHttpHandler(_options);
+
+        context.PostMapRequestHandler += MapRemoteAuthenticationHandler;
+
+        void MapRemoteAuthenticationHandler(object sender, EventArgs e)
+        {
+            var context = ((HttpApplication)sender).Context;
+
+            if (string.Equals(context.Request.Path, _options.AuthenticationEndpointPath)
+                && context.Request.HttpMethod.Equals("GET", StringComparison.OrdinalIgnoreCase))
+            {
+                // TODO : Do we need to even check the API key for this? The handler only returns information
+                //        about the currently authenticated user.
+                if (!string.Equals(_options.RemoteServiceOptions.ApiKey, context.Request.Headers.Get(_options.RemoteServiceOptions.ApiKeyHeader), StringComparison.Ordinal))
+                {
+                    // Using 407 here (proxy authentication required) to differentiate from the scenario of
+                    // a valid API key but no authenticated user.
+                    context.Response.StatusCode = 407;
```

Comment:
It looks like this status code would be blindly returned by the front end to the client. We don't want to send them 407s.
400 bad request or 500 internal server error.

Reply:
500 may be the best option since 400 implies something wrong with the user's request when really it's an issue with proxy setup. I'll use that.

Reply:
Thinking more about this, we don't want any of these responses to get back to the user. I'll leave this as 407 but update the client (`RemoteAppAuthenticationAuthHandler`) to understand 407 and log the issue but not propagate the status code back to the user.